fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/indexes/VersionIndexTest.java

```diff
@@ -2354,4 +2429,37 @@ private <M extends Message> void validateUsingNewerVersionFormat(@Nonnull List<F
         }
         assertFalse(versionKeyPairs.hasNext());
     }
+
+    @Nonnull
+    private List<Tuple> scanIndexToKeys(final RecordQueryPlannerConfiguration.IndexFetchMethod fetchMethod, final String indexName, final ScanProperties direction) throws Exception {
+        if (fetchMethod == RecordQueryPlannerConfiguration.IndexFetchMethod.SCAN_AND_FETCH) {
+            return recordStore.scanIndex(metaData.getIndex(indexName), IndexScanType.BY_VALUE, TupleRange.ALL, null, direction)
+                    .map(IndexEntry::getKey)
+                    .asList().get();
+        } else {
+            IndexScanBounds scanBounds = new IndexScanRange(IndexScanType.BY_VALUE, TupleRange.ALL);
+            KeyExpression primaryKey = recordStore.getRecordMetaData().getRecordType("MySimpleRecord").getPrimaryKey();
+            return recordStore.scanIndexRemoteFetch(indexName, scanBounds, primaryKey, null, direction, IndexOrphanBehavior.ERROR)
```

Comment:
Maybe I'm missing it, but would these tests pass on API versions less than `API_VERSION_7_1`? It looks like they don't include anything gating these tests on the API version.

It also looks like this doesn't actually fallback if the `fetchMethod` is `USE_REMOTE_FETCH_WITH_FALLBACK`. Which maybe is fine, but I'm not sure if that was the intention or not.

Reply:
Good catch - these tests will fail for `API_VERSION` less than `7.1`. With Gradle and CI both at `7.1` we didn't see it but for adopters that set it to `6.3` the test would fail. Added Assumptions on the `API_VERSION`.

In this particular set of tests the index scan by itself does not support FALLBACK (only query does), and so the behavior of FALLBACK is the same as REMOTE_FETCH. We can change the index scan tests to only use two of the fetch modes, which would mean enumerate the two values, which would mean that if we ever add another value to the enum, this test will not cover it. Maybe it is better to use the entire enum (and waste some cycles on redundant runs) and be future-looking?
See change to line 248.

Reply:
Sure, that makes sense. I think having the `_WITH_FALLBACK` and not `_WITH_FALLBACK` methods do the same thing is fine for now. Maybe in the future, if we added some `scanRemoteFetchOrFallBack` method to scan an index, we'd want to switch this over to use that, but that doesn't need to happen now.